lib/Modules/ADHooks/ADHooks.psm1

```diff
@@ -1166,6 +1166,98 @@ function Add-ComputerToADGroupDeprecated {
     }
 }
 
+function Get-FSMORoles {
+    <#
+    .SYNOPSIS
+    Retrieves the FSMO role holders from one or more Active Directory domains and forests.
+    .DESCRIPTION
+    Get-FSMORoles uses the Get-ADDomain and Get-ADForest Active Directory cmdlets to determine
+    which domain controller currently holds each of the Active Directory FSMO roles.
+    .PARAMETER DomainName
+    One or more Active Directory domain names.
+    .EXAMPLE
+    Get-Content domainnames.txt | Get-FSMORoles
+    .EXAMPLE
+    Get-FSMORoles -DomainName domain1, domain2
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromPipeline=$True)]
+        [string[]]$DomainName = $env:USERDOMAIN
+    )
+    BEGIN {
+        Import-Module ActiveDirectory -Cmdlet Get-ADDomain, Get-ADForest -ErrorAction SilentlyContinue
+    }
+    PROCESS {
+        foreach ($domain in $DomainName) {
+            Write-JujuInfo "Querying $domain"
+            Try {
+            $problem = $false
+            $addomain = Get-ADDomain -Identity $domain -ErrorAction Stop
+            } Catch { $problem = $true
+            Write-JujuWarning $_.Exception.Message
+            }
+            if (-not $problem) {
+                $adforest = Get-ADForest -Identity (($addomain).forest)
+
+                return @{
+                    InfrastructureMaster = $addomain.InfrastructureMaster
+                    PDCEmulator = $addomain.PDCEmulator
+                    RIDMaster = $addomain.RIDMaster
+                    DomainNamingMaster = $adforest.DomainNamingMaster
+                    SchemaMaster = $adforest.SchemaMaster
+                }
+            }
+        }
+    }
+}
+
+function Remove-ADComputerFromADForest {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$Computer
+    )
+
+    if($Computer -eq ""){
+        Write-JujuWarning "You need to specify a name to delete"
+        return
+    }
+
+    $cfg = Get-JujuCharmConfig
+    $domainCreds = Get-DomainCredential -UserName $cfg['domain-user'] -Password $cfg['domain-user-password']
+
+    [array]$adminNames = Get-AdministratorAccount
+    $adminName = $adminNames[0]
+    $cfg = Get-JujuCharmConfig
+    $domainCredential = Get-DomainCredential -UserName $adminName -Password $cfg['administrator-password']
+
+    $computerObject = Get-ADComputer -Identity $Computer
+    if($computerObject) {
+        Write-JujuWarning "Removing $Computer from AD domain"
+        $computerObject | Remove-ADObject -Recursive -Confirm:$false -Credential $domainCredential -ErrorAction SilentlyContinue | Out-Null
+    }
+
+    $ZoneName = Get-CharmDomain
+    $DNSResources = $null
+    $DNSResources = Get-DnsServerResourceRecord -ZoneName $ZoneName -Node $Computer -ErrorAction SilentlyContinue
+    if($DNSResources -eq $null){
+        Write-JujuWarning "No DNS record found"
+    } else {
+        foreach ($record in $DNSResources)
```

Comment:
I think this can be rewritten as:

```powershell
$DNSResources | Remove-DnsServerResourceRecord -ZoneName $ZoneName -InputObject $_ -Force -ErrorAction Stop
```

and you can drop the ```for``` loop